src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,74 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { Filter } from '../../types/Filter';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  filter: Filter;
+  setFilter: (prop: Filter) => void;
+  query: string;
+  setQuery: (prop: string) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  filter,
+  setFilter,
+  query,
+  setQuery,
+}) => {
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    switch (event.target.value) {
+      case 'all':
+        setFilter(Filter.All);
+        break;
+      case 'active':
+        setFilter(Filter.Active);
+        break;
+      case 'completed':
+        setFilter(Filter.Completed);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  return (
+    <form
+      onSubmit={event => event.preventDefault()}
+      className="field has-addons"
+    >
+      <p className="control">
+        <span className="select">
+          <select value={filter} onChange={handleChange} data-cy="statusSelect">
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
```

Comment:
map Filter enum

Reply:
Still not fixed 
<img width="850" alt="image" src="https://github.com/mate-academy/react_dynamic-list-of-todos/assets/64750958/6a3a1f74-7f2b-4c43-9ebe-0de991b1d661">